plugins/channel_header_plug/channel_header_plug.tsx

```diff
@@ -277,7 +266,7 @@ class ChannelHeaderPlug extends React.PureComponent<ChannelHeaderPlugProps, Chan
 
         let items = componentItems;
         if (this.props.appsEnabled) {
-            items = componentItems.concat(appBindings.filter((binding) => binding.call).map((binding) => {
+            items = componentItems.concat(appBindings.map((binding) => {
```

Comment:
Before my last commit this was:

```js
items = componentItems.concat(appBindings.filter((binding) => binding.form?.submit).map((binding) => {
```

What are the implications of showing an invalid binding and having an error shown when the user tries to interact with it, or hiding the binding?

This is coming up because I believe the check should really be `binding.form?.submit || binding.submit`, based on the logic in `actions/apps.ts`.

If we want to filter out elements like this here, we should apply the same logic to the post menu items and App Bar items, or have the filtering done in a common function between them.

cc @levb

Reply:
If bindings are filtered on fetch, we can assume all bindings are valid at this level.

We can play safe and do the check here too (and in any other place) but I feel is "safe" to assume the fetch is already filtering these.

Reply:
slightly OT: binding.form.source should also work, perhaps, everywhere?

Reply:
I agree with Lev. Every binding that has a form source, on click, should fetch the form and open it. I recall having some logic round that in place already, but if it is not, we should add it.

Reply:
@levb @larkox One case I found that is not supported is:

`binding.submit` -> call response returns a form with `source` but no `submit`

We just need to have another call to fetch the form in `doAppSubmit` in this case. Wondering if it's worth the code change at this point.